### PR TITLE
Basic media view 

### PR DESCRIPTION
**Pull request by krawieck**
Basic media view, for now supports only images. no galleries, no videos (except gifs)

implemented features:
* hero animation
* pinch to zoom + double tap to zoom
* tap to hide/show controls
* swipe up/down do dismiss
* url share

added widget for easier wrapping of pictures: `FullscreenableImage`

implemented in:
* post
* markdown
* community logo & banner
* instance logo & banner

### CODE REVIEW

File: lib/pages/media_view.dart [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        ]));
+
+    if (showButtons.value) {
+      SystemChrome.setEnabledSystemUIOverlays([
+        SystemUiOverlay.bottom,
+        SystemUiOverlay.top,
+      ]);
+    } else {
+      SystemChrome.setEnabledSystemUIOverlays([]);
+    }
```
Comment (shilangyu) [inline lib/pages/media_view.dart:32]: wrap in useEffect with a dependency

File: lib/pages/media_view.dart [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      body: Dismissible(
+        direction: DismissDirection.vertical,
+        onDismissed: (_) => Navigator.of(context).pop(),
+        key: const Key('media view'),
+        background: Container(color: Colors.black),
+        dismissThresholds: {
+          DismissDirection.vertical: 0,
+        },
+        confirmDismiss: (direction) => Future.value(isZoomedOut.value),
+        resizeDuration: null,
```
Comment (shilangyu) [inline lib/pages/media_view.dart:94]: Is this field required?

File: lib/widgets/markdown_text.dart [hunk 3]
```diff
@@ -26,13 +28,23 @@ class MarkdownText extends StatelessWidget {
...
+            url: uri.toString(),
+            child: CachedNetworkImage(
+              imageUrl: uri.toString(),
+              errorWidget: (context, url, error) => Row(
+                children: [
+                  Icon(Icons.warning),
+                  Text("couldn't load image, ${error.toString()}")
+                ],
+              ),
+            ),
```
Comment (shilangyu) [inline lib/widgets/markdown_text.dart:47]: Doesn't `FullscreenableImage` already wrap in a redirecting `InkWell`?

**Review by shilangyu — CHANGES_REQUESTED** [review]

File: lib/pages/media_view.dart (hunk 2 above)
Comment (krawieck) [reply lib/pages/media_view.dart:95]: yes

**Review by shilangyu — APPROVED** [review]

**Pull request merged**